lib/compat/wordpress-6.3/html-api/class-gutenberg-html-tag-processor-6-3.php

```diff
@@ -1442,20 +1443,22 @@ private function apply_attributes_updates() {
 			$tail_delta = 0;
 
 			foreach ( $this->lexical_updates as $diff ) {
-				$update_head = $bookmark->start >= $diff->start;
-				$update_tail = $bookmark->end >= $diff->start;
-
-				if ( ! $update_head && ! $update_tail ) {
+				if ( $bookmark->start < $diff->start && $bookmark->end < $diff->start ) {
```

Comment:
I still think this would read better as:

```suggestion
				if ( $bookmark->end < $diff->start ) {
```

If `$bookmark->start` is ever after `$bookmark->end`, I'd rather have it fail so we could notice.

Reply:
While we did inline `$update_head` and `$update_tail` here, we didn't otherwise change the condition -- it existed like that prior to this PR. For that reason, I'd like to keep it as is 😬 I know it shouldn't make a difference outside of any pathological cases, but since this PR has been sitting for a while, I'd like to de-risk it as much as possible.

---

Aside: I kinda like that it's symmetrical to the [condition right below](https://github.com/WordPress/gutenberg/pull/47559/files#diff-345dee9d9f8f95c8f166e2959367d9ed7b67853bb3655586168063defb543119R1450):

```php
if ( $bookmark->start >= $diff->start && $bookmark->end < $diff->end ) {
```

I feel it helps me visualize better that the bookmark is either fully past the diff, or enclosed by it.

---

I'm open to changing this in a follow-up though 😊